### PR TITLE
Added confirmation dialog for unlinking a channel

### LINKED ISSUE #3391: Bug: Unlinking a channel from Board's share modal doesn't ask for confirmation
## Steps to reproduce the behavior

1. Create a board.
2. Link it to a channel.
3. Open the board's share modal.
4. Unlink the channel from the modal. Notice there was no confirmation for this action.

## Expected behavior

A confirmation should be asked before unlinking the board.

## Screenshots

https://user-images.githubusercontent.com/18575143/179970866-354936a6-f1d0-42d7-90a8-ccfd14ac1ee7.mp4

### PR DESCRIPTION
**Pull request by harshilsharma63**
Fixes #3391

### CODE REVIEW

File: webapp/src/components/shareBoard/channelPermissionsRow.tsx [hunk 1]
```diff
@@ -43,8 +46,32 @@ const ChannelPermissionsRow = (): JSX.Element => {
         return <></>
     }
 
+    const confirmationDialog = (
+        <ConfirmationDialogBox
+            dialogBox={{
+                heading: intl.formatMessage({
+                    id: 'shareBoard.confirm-unlink.title',
+                    defaultMessage: 'Unlink Channel from Board',
```
Comment (sbishel) [inline webapp/src/components/shareBoard/channelPermissionsRow.tsx:54]:
```suggestion
                    defaultMessage: 'Unlink channel from board',
```

File: webapp/src/components/shareBoard/channelPermissionsRow.tsx [hunk 2]
```diff
@@ -43,8 +46,32 @@ const ChannelPermissionsRow = (): JSX.Element => {
...
+                }),
+                subText: intl.formatMessage({
+                    id: 'shareBoard.confirm-unlink.body',
+                    defaultMessage: 'When you unlink a channel from a board, all members of the channel (existing and new) will loose access to it unless they are given permission separately. {lineBreak} Are you sure you want to unlink it?',
+                }, {
+                    lineBreak: <p/>
+                }),
+                confirmButtonText: intl.formatMessage({
+                    id: 'shareBoard.confirm-unlink.confirmBtnText',
+                    defaultMessage: 'Yes, Unlink',
```
Comment (sbishel) [inline webapp/src/components/shareBoard/channelPermissionsRow.tsx:64]:
```suggestion
                    defaultMessage: 'Yes, unlink',
```

**Review by sbishel — APPROVED** [review]
Just small nit on casing. I've seen @justinegeffen make these changes in the past.

File: webapp/src/components/shareBoard/channelPermissionsRow.tsx (hunk 1 above)
Comment (harshilsharma63) [reply webapp/src/components/shareBoard/channelPermissionsRow.tsx]: Done.

File: webapp/src/components/shareBoard/channelPermissionsRow.tsx (hunk 2 above)
Comment (harshilsharma63) [reply webapp/src/components/shareBoard/channelPermissionsRow.tsx]: Done.

**Pull request merged**